src/main/java/org/eclipse/tractusx/managedidentitywallets/service/HoldersCredentialService.java

```diff
@@ -143,31 +139,41 @@ public PageImpl<VerifiableCredential> getCredentials(String credentialId, String
      * @param callerBpn the caller bpn
      * @return the verifiable credential
      */
-    public VerifiableCredential issueCredential(Map<String, Object> data, String callerBpn) {
-        VerifiableCredential verifiableCredential = new VerifiableCredential(data);
-        Wallet issuerWallet = commonService.getWalletByIdentifier(verifiableCredential.getIssuer().toString());
+    public VerifiableCredential issueCredential(VerifiableCredential verifiableCredential, String callerBpn) {
+        Wallet issuerWallet = commonService.getWalletByDid(new Identifier(verifiableCredential.getIssuer().toString()));
```

Comment:
I see it is a very recurring topic so if it will take too much effort to refactor, we can keep identifier for DID but in that case we should use didOrBpn in places when we both are accepted